### PR TITLE
Adding support for @color and improved font handling for lyrics

### PR DESCRIPTION
**Pull request by rettinghaus**
Verovio now understands: 

* `@color` on `<beatRpt>`, `<hairpin>`, `<octave>`, `<pedal>`, `<rest>`, `<slur>`, `<tie>`, and `<verse>`
* `@lyric.name`, `@lyric.style`, and `@lyric.weight` on `<staffDef>`

### CODE REVIEW

File: src/svgdevicecontext.cpp [hunk 1]
```diff
@@ -170,8 +170,10 @@ void SvgDeviceContext::StartGraphic(Object *object, std::string gClass, std::str
     if (object->HasAttClass(ATT_COLOR)) {
         AttColor *att = dynamic_cast<AttColor *>(object);
         assert(att);
+        if (att->GetColor().length() > 0) {
```
Comment (ahankinson) [inline src/svgdevicecontext.cpp]: The lines below should probably be indented.

File: src/svgdevicecontext.cpp [hunk 2]
```diff
@@ -167,6 +167,15 @@ void SvgDeviceContext::StartGraphic(Object *object, std::string gClass, std::str
         }
     }
 
+    if (object->HasAttClass(ATT_COLOR)) {
+        AttColor *att = dynamic_cast<AttColor *>(object);
+        assert(att);
+        if (att->GetColor().length() > 0) {
```
Comment (lpugin) [inline src/svgdevicecontext.cpp]: Here you should call `att->HasColor()`

File: src/view_floating.cpp [hunk 3]
```diff
@@ -441,9 +441,18 @@ void View::DrawOctave(
         }
     }
     int w, h;
+    int lineWidthFactor = 1;
     std::wstring str;
     str.push_back(code);
-    int lineWidth = m_doc->GetDrawingStemWidth(staff->m_drawingStaffSize);
+    if (octave->HasLwidth()) {
```
Comment (lpugin) [inline src/view_floating.cpp]: clang-format tab length is 4 and not 2

**Comment by rettinghaus** [comment]
@lpugin I formatted view_floating.cpp and changed README. The images folder now may be deleted.

File: include/vrv/pedal.h [hunk 4]
```diff
@@ -18,7 +18,7 @@ namespace vrv {
 // Pedal
 //----------------------------------------------------------------------------
 
-class Pedal : public ControlElement, public TimePointInterface, public AttPedalLog, public AttPlacement {
+class Pedal : public ControlElement, public TimePointInterface, public AttPedalLog, public AttPlacement, public AttColor {
```
Comment (lpugin) [inline include/vrv/pedal.h]: Sorry, I missed this. Att classes should be in alphabetical order

Comment (rettinghaus) [inline include/vrv/pedal.h]: my mistake, I tought I had fixed them all but missed that one!

**Comment by ahankinson** [comment]
Just a minor comment for your next pull request: It would be good if you made a dedicated branch and issue the pull request from there. 

> It's important to use a new branch for pull requests for several reasons:

> It allows you to submit multiple pull requests without confusion. The classical Github gotcha is to 
> continue committing to a pull request branch after making the initial request. When these
> commits are pushed to the remote, they will become part of the original pull request which often 
> ends up conflating unrelated functionality.
>
> When your pull request is merged into the target branch, the maintainer may decide to rebase 
> your commits to avoid a merge commit, or to squash the commits into a single coherent commit. 
> If your pull request was from your 'master' branch, you will encounter problems when merging 
> the target branch back into your own 'master'. Using a temporary branch means it can be 
> discarded once the pull request is accepted and it doesn't matter that your history was rewritten.

From: (Effective Pull Requests and other good practices for teams using GitHub)[http://codeinthehole.com/writing/pull-requests-and-other-good-practices-for-teams-using-github/]

**Comment by lpugin** [comment]
Thanks. @rettinghaus could you an example for the line parameters to http://www.verovio.org/features.xhtml?id=octave-shift ? You can make a pull request to gh-pages-develop

I am not sure where to put an example for `@color`. Any ideas?

**Comment by rettinghaus** [comment]
Yeah, I'll add some examples.

**Comment by rettinghaus** [comment]
@lpugin is there a hires version of the Verovio logo?

**Comment by lpugin** [comment]
[verovio-logo.zip](https://github.com/rism-ch/verovio/files/510501/verovio-logo.zip)

**Pull request merged**